tests/conftest.py

```diff
@@ -191,20 +212,20 @@ def format_pyproject_toml(deps: Deps, extras: ExtraDeps) -> str:
         )
 
     def format_deps(
-        filename: str, all_deps: Union[Deps, Tuple[Deps, ExtraDeps]]
+        filename: str, all_deps: Union[Deps, Tuple[Deps, ExtraDeps]], is_dynamic: bool
     ) -> str:
         if isinstance(all_deps, tuple):
             deps, extras = all_deps
         else:
             deps, extras = all_deps, {}
-        formatters: Dict[str, Callable[[Deps, ExtraDeps], str]] = {
+        formatters: Dict[str, Callable[[Deps, ExtraDeps, bool], str]] = {
```

Comment:
> @jherland, was there any reason we preferred originally using assertions and having a unified interface for these formatters?

I think it comes down to the formatter selection on line 227, and the subsequent call on line 228. Since we do a single call for the chosen formatter (whichever it ends up being), they must have a single, unified interface (i.e. at the point we make the call we no longer know/care which formatter we call).

We _could_ attempt to diversify the interface, and then use `functools.partial` to prepare the different argument lists for each formatter. I think it would look something like this:

```python
from functools import partial
...
        formatters: Dict[str, Callable[[], str]] = {
            "requirements.txt": partial(format_requirements_txt, deps),  # default choice
            "setup.py": partial(format_setup_py, deps, extras),
            "setup.cfg": partial(format_setup_cfg, deps),
            "pyproject.toml": partial(format_pyproject_toml, deps extras, is_dynamic),
        }
        formatter = formatters.get(Path(filename).name, formatters["requirements.txt"])
        return formatter()
```

but I'm not sure this is really any better:
- `partial` functions are often harder to read/understand,
- I don't like the duplication of args inside the `formatters` dict,
- we lose the asserts that would catch misuse of `extras`/`is_dynamic` with formatters that don't support them.

Reply:
Your analysis makes sense. I didn't follow up my logic up until line 228. I am closing this. Let's stick with the unified interface.